solidity/contracts/system/TBTCSystem.sol

```diff
@@ -142,13 +144,61 @@ contract TBTCSystem is Ownable, ITBTCSystem, DepositLog {
             _keepThreshold,
             _keepSize
         );
+        tbtcToken = _tbtcToken;
         setTbtcDepositToken(_tbtcDepositToken);
-        _initialized = true;
+        initializedTimestamp = block.timestamp;
         allowNewDeposits = true;
     }
 
-    /// @notice gets whether new deposits are allowed.
-    function getAllowNewDeposits() external view returns (bool) { return allowNewDeposits; }
+    /// @notice returns whether new deposits should be allowed.
+    function getAllowNewDeposits() external view returns (bool) {
+        if (initializedTimestamp == 0 || !allowNewDeposits) {
+            return false;
+        }
+
+        uint256 supply = tbtcToken.totalSupply();
```

Comment:
We can probably access `tbtcToken` directly on the vending machine, fwiw. No strong opinion on whether that's better or worse… It seems better encapsulated to consider the vending machine the source of truth on supply from `TBTCSystem`'s perspective, or even the ultimate arbiter of permitted supply. One scenario falls out from this line of reasoning…

Say there are 120 BTC under custody in the first month, and the 100th TBTC is minted. This hits the supply cap. `TBTCSystem` disallows further deposits until some TBTC is taken out of circulation. Should the vending machine still allow the minting of the remaining 20 BTC into TBTC?

If the answer is no, then perhaps what we need here is for the vending machine to track the permitted TBTC supply, refuse to mint further TBTC when the supply cap is reached, and expose e.g. `isTBTCMintingAllowed` or similar. Then `TBTCSystem` uses that function, which encapsulates the vending machine's assessment of the supply cap, as the additional factor here. This has the additional effect of removing the direct `TBTCToken` dependency from `TBTCSystem`.

Reply:
Downsides of doing this in the vending machine:

* What we actually want is to stop new deposits to limit risk, not just stop minting. This is a close proxy without wider system changes
* The dApp would need to check the vending machine and refuse new deposits of the supply cap was hit

Upsides

* Cleaner implementation, perhaps, and nice to not add a token contract reference?

I think the best solution without additional bookkeeping might be "both"- stopping new deposits and additional minting. Unclear to me whether it's worth the effort though, and I want to make as few modifications as possible.

Reply:
That’s what I meant: having the vending machine guard minting and expose the current stare through a public function, and having TBTCSystem use that public function with the reference it already has to guard deposits as well.